tools/model_tools/src/openvino/model_zoo/open_model_zoo.py

```diff
@@ -0,0 +1,217 @@
+            omz_converter.converter(['--name=' + model_name, '--precisions=' + precision,
+                        '--download_dir=' + str(download_dir)])
+        return cls(name, model_path, description, task_type, subdirectory, ie)
+
+    @classmethod
+    def from_pretrained(cls, model_path, *, task_type=None, ie=None):
+        '''
+        Loads model from existing .xml, .onnx files.
+        Parameters
+        ----------
+            model_path
+                Path to .xml or .onnx file.
+            task_type
+                Type of task that the model performs.
+            ie
+                Inference Engine instance
+        '''
+        model_path = Path(model_path).resolve()
+
+        if not model_path.exists():
+            raise ValueError('Path {} to model file does not exist.'.format(model_path))
+
+        if model_path.suffix not in ['.xml', '.onnx']:
+            raise ValueError('Unsupported model format {}. Only .xml or .onnx supported.'.format(model_path.suffix))
+
+        description = 'Pretrained model {}'.format(model_path.name)
+        name = model_path.stem
+        model = cls._load_model(cls, name)
+        task_type = model.task_type if model else task_type
+        subdirectory = model.subdirectory if model else model_path.parent
+        model_path = str(model_path)
+
+        return cls(name, model_path, description, task_type, subdirectory, ie)
+
+    def _load_model(self, model_name):
+        parser = argparse.ArgumentParser()
+        args = argparse.Namespace(all=False, list=None, name=model_name, print_all=False)
+        try:
+            model = _configuration.load_models_from_args(parser, args, _common.MODEL_ROOT,
+                        mode=_configuration.ModelLoadingMode.ignore_composite)[0]
+        except SystemExit:
+            model = None
+
+        return model
+
+    def _load(self, device):
+        self.net = self.ie.read_model(self.model_path)
+        self.exec_net = self.ie.compile_model(self.net, device)
+        self.device = device
+
+    def __call__(self, inputs, device='CPU'):
+        if self.ie is None:
+            raise TypeError('ie is not specified or is of the wrong type.'
+                            'Please check ie is of openvino.runtime.Core type.')
+
+        if self.exec_net is None or device != self.device:
+            self._load(device)
+
+        input_names = [input.get_any_name() for input in self.net.inputs]
+        infer_request = self.exec_net.create_infer_request()
+
+        for input_name in inputs:
+            if input_name not in input_names:
+                raise ValueError('Unknown input name {}'.format(input_name))
+
+        output_names = [output.get_any_name() for output in self.net.outputs]
+        infer_request.infer(inputs)
+        result = {name: infer_request.get_tensor(name).data for name in output_names}
+        return result
+
+    @property
+    def accuracy_checker_config(self):
+        if self._accuracy_config is None:
```

Comment:
as I can see, configs are read only properties, why do you need to cache them in class if they can not be changed?
yaml reading is not so time consuming operation for caching it

Reply:
Removed config caching

> I can understand, why you need has model config in memory, it is required for access to some fields (input_shapes, layouts e.t.c.), but accuracy checker config is not so informative

There were some requests to provide a way to access info from accuracy checker configs.